keanu-project/src/main/java/io/improbable/keanu/tensor/generic/GenericTensor.java

```diff
@@ -22,6 +22,18 @@
     private long[] shape;
     private long[] stride;
 
+    public static <T> GenericTensor<T> create(T data, long[] shape) {
+        if (Arrays.equals(shape, Tensor.SCALAR_SHAPE)) {
```

Comment:
I don't think you need this branch - you can always `return new GenericTensor<>(data, shape);`